### PR TITLE
Convert Object tests to table-driven style

### PR DESCRIPTION
**Pull request by jeremyphua**
#386

### CODE REVIEW

**Comment by jeremyphua** [comment]
Hi @gavv , PR is ready. Let me know if there are any changes to be done!

File: object_test.go [hunk 1]
```diff
@@ -457,210 +432,212 @@ func TestObject_IsEqual(t *testing.T) {
...
-		}{Foo: 123.00})
-		value.chain.assert(t, failure)
-		value.chain.clear()
+				NewObject(reporter, tc.value).NotInList(tc.testValue...).
+					chain.assert(t, !tc.wantEqual)
+			})
+		}
 	})
 
 	t.Run("empty", func(t *testing.T) {
```
Comment (gavv) [inline object_test.go:540]: I think this "empty" can be merged into "basic" table above.

**Review by gavv — COMMENTED** [review]
Thanks for PR!

**Comment by jeremyphua** [comment]
I've made the changes according to your comments. Let me know if I missed out some stuff! @gavv

**Comment by gavv** [comment]
Thanks!

**Comment by gavv** [comment]
Also welcome to chat https://discord.gg/5SCPCuCWA9

**Pull request merged**